### PR TITLE
Add sanction check

### PR DESCRIPTION
**Pull request by ae2079**

### CODE REVIEW

File: src/components/Controllers/UserController.tsx [hunk 1]
```diff
@@ -91,7 +95,29 @@ export const UserController = () => {
     };
   }, []);
 
-  return showSignModal ? (
+  useEffect(() => {
+    // Check if the address is in the sanction list
+    const checkSanctionStatus = async () => {
```
Comment (aminlatifi) [inline src/components/Controllers/UserController.tsx:100]: As API has a limited usage cap, it's better to use `useQuery` and utilize its caching feature

**Comment by aminlatifi** [comment]
Also, it's better to use the API key ASAP, since with a high chance our testers will hit the 100 per day limit.

**Review by MohammadPCh — APPROVED** [review]
LGTM, Thanks @ae2079

**Comment by ae2079** [comment]
> Also, it's better to use the API key ASAP, since with a high chance our testers will hit the 100 per day limit.

based on adding useQuery, if the use same address and not refresh the page 100 times in a day, they should not reach the limit, also in case of failure, we let them continue and not block them, so it should not be a major problem, and in the giveth we use this same approach (even without useQuery), so it should be fine

**Pull request merged**